hs_core/search_indexes.py

```diff
@@ -711,67 +692,82 @@ def prepare_geometry_type(self, obj):
+            self._cache_queryset(obj, 'cached_geofeature_logical_files', obj.geofeaturelogicalfile_set.all())
+            for f in obj.cached_geofeature_logical_files:
+                field_info = f.metadata.fieldinformations.all().first()
+                if field_info is not None and field_info.fieldTypeCode is not None:
+                    return field_info.fieldTypeCode.strip()
         return None
 
     def prepare_variable(self, obj):
         """
         Return metadata variable names if exists, otherwise return empty array.
         """
+
         variables = set()
-        for f in obj.netcdflogicalfile_set.all():
-            for v in f.metadata.variables.all():
-                if discoverable(v.name):
-                    variables.add(v.name.strip())
-        for f in obj.timeserieslogicalfile_set.all():
-            for v in f.metadata.variables:
-                # TODO: inconsistent use of variable code and variable name
-                if discoverable(v.variable_name):
-                    variables.add(v.variable_name.strip())
-        for f in obj.reftimeserieslogicalfile_set.all():
-            for v in f.metadata.variables:
-                # TODO: inconsistent use of variable code and variable name
-                if discoverable(v.name):
-                    variables.add(v.name.strip())
-        for f in obj.georasterlogicalfile_set.all():
-            for b in f.metadata.bandInformations:
-                if discoverable(b.variableName):
-                    variables.add(b.variableName)
+        if obj.resource_type == 'CompositeResource':
+            self._cache_queryset(obj, 'cached_netcdf_logical_files', obj.netcdflogicalfile_set.all())
+            for f in obj.cached_netcdf_logical_files:
+                for v in f.metadata.variables.all():
+                    if discoverable(v.name):
+                        variables.add(v.name.strip())
+
+            self._cache_queryset(obj, 'cached_timeseries_logical_files', obj.timeserieslogicalfile_set.all())
+            for f in obj.cached_timeseries_logical_files:
+                for v in f.metadata.variables:
+                    # TODO: inconsistent use of variable code and variable name
+                    if discoverable(v.variable_name):
+                        variables.add(v.variable_name.strip())
+            self._cache_queryset(obj, 'cached_reftimeseries_logical_files', obj.reftimeserieslogicalfile_set.all())
+            for f in obj.cached_reftimeseries_logical_files:
+                for v in f.metadata.variables:
+                    # TODO: inconsistent use of variable code and variable name
+                    if discoverable(v.name):
+                        variables.add(v.name.strip())
+            for f in obj.georasterlogicalfile_set.all():
+                for b in f.metadata.bandInformations:
+                    if discoverable(b.variableName):
+                        variables.add(b.variableName)
         return list(variables)
```

Comment:
return None if not a Composite resource?
same question for all of the remaining functions in this Class -- should they default return None?

Reply:
It seems we need to return an empty list when more than one item is expected.

Reply:
Ah nice, okay!